aikau/src/main/resources/alfresco/core/_PublishOrLinkMixin.js

```diff
@@ -0,0 +1,148 @@
+ * or navigate to a page.</p>
+ *
+ * <p>By providing a publishTopic in the config, the widget will publish to that topic when clicked
+ * (publishPayload, publishGlobal and publishToParent are also supported).</p>
+ *
+ * <p>By providing a targetUrl (and optional targetUrlType), the widget will act as a link when clicked</p>
+ *
+ * <p>This mixin derives its functionality from two other mixins in order to provide a simple way of
+ * decorating a widget to add "clicking functionality". Please note though that the enhanced capabilities of
+ * [_HtmlAnchorMixin]{@link module:alfresco/navigation/_HtmlAnchorMixin} and
+ * [_PublishPayloadMixin]{@link module:alfresco/renderers/_PublishPayloadMixin} are also available to
+ * anyone mixing in this class.</p>
+ *
+ * <p><em>NOTE:</em> This widget's test cases are fulfilled by the Logo test page</p>
+ *
+ * @module alfresco/core/_PublishOrLinkMixin
+ * @extends module:alfresco/navigation/_HtmlAnchorMixin
+ * @mixes module:alfresco/renderers/_PublishPayloadMixin
+ * @author Martin Doyle
+ * @since 1.0.40
+ */
+define(["alfresco/navigation/_HtmlAnchorMixin",
+        "alfresco/renderers/_PublishPayloadMixin",
+        "dojo/_base/declare",
+        "dojo/_base/lang",
+        "dojo/dom-class",
+        "dojo/keys",
+        "dojo/on"],
+        function(_HtmlAnchorMixin, _PublishPayloadMixin, declare, lang, domClass, keys, on) {
+
+   return declare([_HtmlAnchorMixin, _PublishPayloadMixin], {
+
+      /**
+       * An array of the CSS files to use with this widget.
+       * 
+       * @instance
+       * @type {object[]}
+       * @default [{cssFile:"./css/_PublishOrLinkMixin.css"}]
+       */
+      cssRequirements: [{cssFile:"./css/_PublishOrLinkMixin.css"}],
+
+      /**
+       * When set to true, the widget/anchors will be removed from the page tab-order by setting the tabindex to -1.
+       *
+       * @instance
+       * @override
+       * @see module:alfresco/navigation/_HtmlAnchorMixin#excludeFromTabOrder
+       * @type {boolean}
+       * @default
+       */
+      excludeFromTabOrder: false,
+
+      /**
+       * The title text to put on the generated link(s).
+       *
+       * @instance
+       * @override
+       * @type {string}
+       * @default
+       */
+      label: null,
+
+      /**
+       * This is the URL to navigate to when the widget is clicked.
+       *
+       * @instance
+       * @type {string}
+       * @default
+       */
+      targetUrl: null,
+
+      /**
+       * Indicates how the target URL should be handled. This defaults to "PAGE_RELATIVE" which means that the URL
+       * will be appended to the 'AlfConstants.URL_PAGECONTEXT' Global JavaScript constant. This can be overridden
+       * on instantiation to indicate that another URL type, such as "FULL_PATH" should be used.
+       *
+       * @instance
+       * @type {string}
+       * @default
+       */
+      targetUrlType: "PAGE_RELATIVE",
+
```

Comment:
The _HtmlAnchorMixin doesn't support that functionality. Should it? Thought we were just putting it in as-is initially?